VideoWeb/VideoWeb/ClientApp/src/app/services/user-media-stream.service.ts

```diff
@@ -80,45 +80,47 @@ export class UserMediaStreamService {
                 this.currentStream = this.mediaStreamService.initialiseNewStream(cameraStream.getVideoTracks());
                 this.activeCameraStream = cameraStream;
 
-                this.mediaStreamService
-                    .getStreamForMic(microphoneDevice)
-                    .pipe(take(1))
-                    .subscribe(microphoneStream => {
-                        this.logger.debug(`${this.loggerPrefix} Got microphone stream`, {
-                            mic: microphoneDevice,
-                            stream: microphoneStream,
-                            isAudioOnly: this.isAudioOnly
-                        });
-
-                        microphoneStream.getAudioTracks().forEach(track => {
-                            this.currentStream.addTrack(track);
-                        });
-
-                        this.activeMicrophoneStream = microphoneStream;
-
-                        this.logger.info(`${this.loggerPrefix} Built initial stream`, {
-                            mic: microphoneDevice,
-                            cam: cameraDevice,
-                            stream: this.currentStream,
-                            tracks: this.currentStream.getTracks(),
-                            isAudioOnly: this.isAudioOnly
-                        });
-
-                        this.currentStreamSubject.next(this.currentStream);
-
-                        this.logger.debug(`${this.loggerPrefix} Subscribing to active video and microphone device changes`);
-                        this.userMediaService.activeVideoDevice$.pipe(skip(1)).subscribe(videoDevice => {
-                            this.onActiveCameraChanged(videoDevice);
-                        });
-
-                        this.userMediaService.activeMicrophoneDevice$.pipe(skip(1)).subscribe(activeMicrophoneDevice => {
-                            this.onActiveMicrophoneChanged(activeMicrophoneDevice);
-                        });
-
-                        this.userMediaService.isAudioOnly$.subscribe(audioOnly => {
-                            this.onIsAudioOnlyChanged(audioOnly);
+                if (!this.activeMicrophoneStream) {
+                    this.mediaStreamService
+                        .getStreamForMic(microphoneDevice)
+                        .pipe(take(1))
+                        .subscribe(microphoneStream => {
```

Comment:
if you're refactoring this, maybe look at using a `forkJoin` as to avoid subscribing inside another subscriber

Reply:
Reworked to be a lot simpler. The initialisation can be done with the existing event handlers and not need a separate path to initialise the current stream.